### PR TITLE
Aim api integration tests Search Metrics endpoint

### LINKED ISSUE #354: Add integration tests for Aim API SearchMetrics endpoint

### PR DESCRIPTION
**Pull request by fabiovincenzi**
Fixes #354

### CODE REVIEW

**Review by suprjinx — DISMISSED** [review]
Nice work -- a very difficult test!

File: tests/integration/golang/aim/metric/search_test.go [hunk 1]
```diff
@@ -34,97 +41,748 @@ func (s *SearchMetricsTestSuite) SetupTest() {
...
+						RunID:     run.ID,
+						LastIter:  int64(decodedData[itersKey].([]float64)[0]),
+					}
+					decodedMetrics = append(decodedMetrics, &m)
+					metricCount++
+				}
+			}
+
+			// Check if the received metrics match the expected ones
+			for _, metric := range metrics {
```
Comment (dsuhinin) [inline tests/integration/golang/aim/metric/search_test.go:779]: can't understand why we can't just compare `tt.metrics` and `decodedMetrics`? can you please explain why do we need `metrics`?

Comment (fabiovincenzi) [reply tests/integration/golang/aim/metric/search_test.go:779]: You're right, I was doing this to check if the not expected ones weren't in the response but yes we can check it by comparing `tt.metrics` and `decodedMetrics`, I'll fix this, thanks!

Comment (dsuhinin) [reply tests/integration/golang/aim/metric/search_test.go:779]: but you can create a helper function to apply smart comparing: length is equal, all elements existing in both slices, etc.

File: tests/integration/golang/aim/metric/search_test.go [hunk 2]
```diff
@@ -34,97 +41,748 @@ func (s *SearchMetricsTestSuite) SetupTest() {
...
 	for _, tt := range tests {
 		s.T().Run(tt.name, func(T *testing.T) {
-			data, err := s.client.DoStreamRequest(
+			var resp []byte
+			query, err := urlquery.Marshal(tt.request)
+			assert.Nil(s.T(), err)
+			resp, err = s.client.DoStreamRequest(
 				http.MethodGet,
-				fmt.Sprintf("/runs/search/metric?%s", tt.query),
+				fmt.Sprintf(`/runs/search/metric?%s`, query),
```
Comment (jgiannuzzi) [inline tests/integration/golang/aim/metric/search_test.go:747]: please use proper query string encoding by building a `net.url.URL` struct

Comment (jgiannuzzi) [reply tests/integration/golang/aim/metric/search_test.go:747]:
```
$ grep -rE 'fmt\.Sprintf\("/.*\?' tests                    
tests/integration/golang/aim/experiment/get_experiment_runs_test.go:            fmt.Sprintf("/experiments/%d/runs?limit=4&offset=%s", *experiment.ID, runs[8].ID),
tests/integration/golang/aim/run/search_test.go:                                fmt.Sprintf("/runs/search/run?%s", query),
tests/integration/golang/aim/run/archive_batch_test.go:                         fmt.Sprintf("/runs/archive-batch?archive=%s", tt.archiveParam),
tests/integration/golang/aim/metric/search_test.go:                             fmt.Sprintf("/runs/search/metric?%s", tt.query),
```

Comment (dsuhinin) [reply tests/integration/golang/aim/metric/search_test.go:747]: yep, we can improve it a little bit and pass for example object, which could contain `body` parameter or `query` parameters and under the hood client will attach/serialise body or build query string.

Comment (jgiannuzzi) [reply tests/integration/golang/aim/metric/search_test.go:747]: I'll create an issue to deal with it

Comment (dsuhinin) [reply tests/integration/golang/aim/metric/search_test.go:747]: sure! I can create a concept then.

Comment (jgiannuzzi) [reply tests/integration/golang/aim/metric/search_test.go:747]: https://github.com/G-Research/fasttrackml/issues/393 — assigned to you @dsuhinin

Comment (jgiannuzzi) [reply tests/integration/golang/aim/metric/search_test.go:747]: if you start working on this, please target `features/namespaces`!

Comment (dsuhinin) [reply tests/integration/golang/aim/metric/search_test.go:747]: yep yep. already!

**Pull request closed without merging**